### PR TITLE
Pass deno CLI args to functions serve

### PR DESCRIPTION
**Pull request by kruttik-lab49**
## What kind of change does this PR introduce?

feature

## What is the current behavior?

Currently, there is no way to pass deno cli arguments to docker containers.

See discussion thread [here](https://github.com/supabase/supabase/discussions/10989).

## What is the new behavior?

This PR adds an ability to pass deno cli arguments to `supabase functions serve` command

## Additional context

Most of the details are in the discussion forum above.

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1 @@
+golang 1.19.3
```
Comment (sweatybridge) [inline .tool-versions:1]: Is there any preference for this over the current version specified in https://github.com/supabase/cli/blob/main/go.mod#L3?

File: .gitignore [hunk 2]
```diff
@@ -20,3 +20,9 @@
 # NPM
 node_modules
 package-lock.json
+
+# Ignore built artifact - go build
+cli
+
+#Ignore IDE
+.idea
```
Comment (sweatybridge) [inline .gitignore:28]: let's group this under line 18 please

**Review by sweatybridge — COMMENTED** [review]
A few nitpicks and general comments:

Is exposing all deno args necessary? As the CLI frequently upgrades deno version to match that used by the platform, I think it will be harder to maintain compatibility if we allow arbitrary deno flags.

How about exposing only `SUPABASE_DENO_CERT` env var which will be injected as `DENO_CERT` inside the container? This won't conflict with host system deno envs and minimizes the compatibility surface.

**Comment by sweatybridge** [comment]
Anyway thanks for the thorough investigation. I've released a fix to pass user defined env vars from `--env-file` flag to all deno commands. Let me know if it solves your issue.

**Comment by kruttik-lab49** [comment]
> Anyway thanks for the thorough investigation. I've released a fix to pass user defined env vars from `--env-file` flag to all deno commands. Let me know if it solves your issue.

Great!! I will give it a try and report back.

**Comment by kruttik-lab49** [comment]
I can confirm that it works!! Thank you so much.

**Pull request closed without merging**